### PR TITLE
User story7

### PR DESCRIPTION
**Pull request by zachlp2016**
Added registration user story for new user.
Added sad path testing to verify emails are unique.

### CODE REVIEW

**Comment by VinceCarollo** [comment]
I'm seeing an error if two different passwords are put in. Error: "Render and/or redirect were called multiple times in this action." Other than that it looks good. We should add some flash notices at a later date.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  end
+
+  def create
+    @user = User.new(strong_params)
+    if password_confirmation != true
+      render :new
+      flash[:notice] = "Those passwords don't match"
+    end
+    if @user.save
```
Comment (BrennanAyers) [inline app/controllers/users_controller.rb]: I think this needs to be an `elsif`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  def password_confirmation
+    if params["user"]["password"] == params["user"]["confirm_password"]
+      true
+    else
+      false
+    end
+  end
+
+  def strong_params
```
Comment (BrennanAyers) [inline app/controllers/users_controller.rb:41]: We should probably just keep to the naming convention of *_params, in this case `user`

File: app/models/user.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class User < ApplicationRecord
   has_many :items
   has_many :orders
-  
-  validates_presence_of :password_digest, :role, :active, :name, :address, :city, :state, :zip
+
+  validates_presence_of :password_digest, :name, :address, :city, :state, :zip
```
Comment (BrennanAyers) [inline app/models/user.rb:5]: @VinceCarollo What was the validation you were using for `:role` ?

**Review by VinceCarollo — APPROVED** [review]

**Review by BrennanAyers — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (VinceCarollo) [reply app/models/user.rb]: validates_inclusion_of :active, in: [true, false]

Comment (VinceCarollo) [reply app/models/user.rb:5]: validates_inclusion_of :active, in: [true, false]

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    if password_confirmation != true
+      render :new
+      flash[:notice] = "Those passwords don't match"
+    end
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:notice] = "You are now registered and logged in."
+      redirect_to '/profile'
+    else
+      render :new
```
Comment (BrennanAyers) [inline app/controllers/users_controller.rb:19]: Might be worth adding a flash message in here that says there is an error with their form/our servers

Comment (WHomer) [reply app/controllers/users_controller.rb:19]: @BrennanAyers opened an issue #91

**Review by BrennanAyers — APPROVED** [review]

**Pull request merged**